bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/handler/FreeAtHomeDeviceHandler.java

```diff
@@ -0,0 +1,618 @@
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.thing.type.AutoUpdatePolicy;
+import org.openhab.core.thing.type.ChannelKind;
+import org.openhab.core.thing.type.ChannelType;
+import org.openhab.core.thing.type.ChannelTypeBuilder;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.StateDescriptionFragmentBuilder;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.FrameworkUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link FreeAtHomeDeviceHandler} is responsible for handling the generic free@home device main communication
+ * and thing updates
+ *
+ * @author Andras Uhrin - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class FreeAtHomeDeviceHandler extends BaseThingHandler implements FreeAtHomeDeviceStateListener {
+
+    private static final String CHANNEL_URI = "channel-type:freeathomesystem:config";
+    private static final String DEVICE_ID_CONFIG = "deviceId";
+
+    private final Logger logger = LoggerFactory.getLogger(FreeAtHomeDeviceHandler.class);
+    private FreeAtHomeDeviceDescription device = new FreeAtHomeDeviceDescription();
+    private FreeAtHomeChannelTypeProvider channelTypeProvider;
+    private TranslationProvider i18nProvider;
+    private Locale locale;
+    private Bundle bundle;
+
+    private Map<ChannelUID, FreeAtHomeDatapointGroup> mapChannelUID = new HashMap<ChannelUID, FreeAtHomeDatapointGroup>();
+    private Map<String, ChannelUID> mapEventToChannelUID = new HashMap<String, ChannelUID>();
+
+    public FreeAtHomeDeviceHandler(Thing thing, FreeAtHomeChannelTypeProvider channelTypeProvider,
+            TranslationProvider i18nProvider, LocaleProvider localeProvider) {
+        super(thing);
+
+        this.channelTypeProvider = channelTypeProvider;
+        this.i18nProvider = i18nProvider;
+        this.bundle = FrameworkUtil.getBundle(getClass());
+        this.locale = localeProvider.getLocale();
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+
+        scheduler.execute(() -> {
+            final String locDeviceId = (String) getConfig().get(DEVICE_ID_CONFIG);
+
+            Bridge bridge = this.getBridge();
+
+            if (bridge != null) {
+                ThingHandler handler = bridge.getHandler();
+
+                if (handler instanceof FreeAtHomeBridgeHandler bridgeHandler) {
+                    if (locDeviceId != null) {
```

Comment:
You could also validate that it's filled out, i.e. `!locDeviceId.isBlank()`. You could consider providing a small DTO for deserializing the configuration, like this:

https://github.com/openhab/openhab-addons/blob/9ad3c66e17bb0be2ca9a42ccad6861749ba21ed8/bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/config/Clip2ThingConfig.java#L23-L25

for corresponding declaration:

https://github.com/openhab/openhab-addons/blob/a7d183e2f6e6eb7cf6157475eb464bb86f2b71a8/bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/Clip2Thing.xml#L110-L115

and by that simplifying the handler code (can no longer be null because defaulted to empty string):

https://github.com/openhab/openhab-addons/blob/a7d183e2f6e6eb7cf6157475eb464bb86f2b71a8/bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java#L604-L609

Reply:
I see now that you did this for the bridge handler, so it would be good to align approach.

Reply:
**Fixed**

Same solution in the gateway and in the thing handler